lib/connection.js

```diff
@@ -120,8 +121,14 @@ class Connection extends EventEmitter {
     let name, version;
     if (typeof app === 'string') {
       [name, version] = app.split('@');
-      if (version && !semver.valid(version)) {
-        throw new Error('Invalid semver version');
+      if (version && !semver.validRange(version)) {
```

Comment:
Also, you should add a check for the version provided inside of an object, now you just assign its value even if it's incorrect (on line 135).